diff --git a/src/system/emscripten/system.c b/src/system/emscripten/system.c
index 12269ae0..840a7582 100644
--- a/src/system/emscripten/system.c
+++ b/src/system/emscripten/system.c
@@ -13,6 +13,7 @@
 //
 
 #include <emscripten/html5.h>
+#include <unistd.h>
 #include <errno.h>
 #include <pthread.h>
 #include <stddef.h>
@@ -124,12 +125,12 @@ z_result_t _z_condvar_wait_until(_z_condvar_t *cv, _z_mutex_t *m, const z_clock_
 
 /*------------------ Sleep ------------------*/
 z_result_t z_sleep_us(size_t time) {
-    emscripten_sleep((time / 1000) + (time % 1000 == 0 ? 0 : 1));
+    usleep((useconds_t)time);
     return 0;
 }
 
 z_result_t z_sleep_ms(size_t time) {
-    emscripten_sleep(time);
+    usleep((useconds_t)time * 1000);
     return 0;
 }
 
