diff --git a/3rdparty/Qt-Advanced-Docking/CMakeLists.txt b/3rdparty/Qt-Advanced-Docking/CMakeLists.txt
index 6b650f1b..67ac7a9f 100644
--- a/3rdparty/Qt-Advanced-Docking/CMakeLists.txt
+++ b/3rdparty/Qt-Advanced-Docking/CMakeLists.txt
@@ -67,7 +67,9 @@ target_link_libraries(qt_advanced_docking PUBLIC Qt5::Core Qt5::Gui Qt5::Widgets

 if(UNIX AND NOT APPLE)
     target_link_libraries(qt_advanced_docking PUBLIC Qt5::X11Extras)
-        target_link_libraries(qt_advanced_docking PRIVATE xcb)
+    find_package(XCB REQUIRED)
+    target_link_libraries(qt_advanced_docking PRIVATE ${XCB_LIBRARIES})
+    target_include_directories(qt_advanced_docking SYSTEM PUBLIC ${XCB_INCLUDE_DIRS})
 endif()

 set_target_properties(qt_advanced_docking PROPERTIES
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 385b7899..b1d6f2ab 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -114,11 +114,9 @@ find_package(Qt5 REQUIRED COMPONENTS
 set( QT_LINK_LIBRARIES
     Qt5::Core
     Qt5::Widgets
-    Qt5::PrintSupport
     Qt5::Xml
     Qt5::Concurrent
     Qt5::Svg
-    Qt5::OpenGL
     Qt5::WebSockets
     )

