diff --git a/CMakeLists.txt b/CMakeLists.txt
index b8dd549bcb..98737c501a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,6 +18,8 @@ find_package(Eigen3 REQUIRED)
 find_package(octomap REQUIRED)
 find_package(geometric_shapes REQUIRED)
 find_package(tf2_ros REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(Boost REQUIRED)
 
 include_directories(include)
 include_directories(SYSTEM ${EIGEN3_INCLUDE_DIRS} ${X11_INCLUDE_DIR})
@@ -38,14 +40,27 @@ add_library(
 set_target_properties(
   moveit_ros_occupancy_map_monitor
   PROPERTIES VERSION "${moveit_ros_occupancy_map_monitor_VERSION}")
-ament_target_dependencies(moveit_ros_occupancy_map_monitor
-                          ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(
+  moveit_ros_occupancy_map_monitor
+  rclcpp::rclcpp
+  moveit_core::moveit_core
+  ${moveit_msgs_TARGETS}
+  pluginlib::pluginlib
+  octomap
+  geometric_shapes::geometric_shapes
+  Boost::headers)
 
 add_executable(moveit_ros_occupancy_map_server src/occupancy_map_server.cpp)
-ament_target_dependencies(moveit_ros_occupancy_map_server PUBLIC
-                          ${THIS_PACKAGE_INCLUDE_DEPENDS})
-target_link_libraries(moveit_ros_occupancy_map_server
-                      PRIVATE moveit_ros_occupancy_map_monitor)
+target_link_libraries(
+  moveit_ros_occupancy_map_server
+  PRIVATE moveit_ros_occupancy_map_monitor
+          rclcpp::rclcpp
+          moveit_core::moveit_core
+          ${moveit_msgs_TARGETS}
+          pluginlib::pluginlib
+          octomap
+          geometric_shapes::geometric_shapes
+          Boost::headers)
 
 install(
   TARGETS moveit_ros_occupancy_map_monitor
