diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5b784cc..eb31e62 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,6 +14,7 @@ include(CheckLibraryExists)
 
 find_package(ament_cmake REQUIRED)
 find_package(ament_cmake_ros_core REQUIRED)
+find_package(Threads REQUIRED)
 
 if(UNIX AND NOT APPLE)
   include(cmake/check_c_compiler_uses_glibc.cmake)
@@ -87,6 +88,8 @@ if(BUILD_TESTING AND NOT RCUTILS_DISABLE_FAULT_INJECTION)
 endif()
 
 target_link_libraries(${PROJECT_NAME} ${CMAKE_DL_LIBS})
+target_link_libraries(${PROJECT_NAME} Threads::Threads)
+ament_export_dependencies(Threads)
 
 check_library_exists(atomic __atomic_load_8 "" HAVE_LIBATOMICS)
