diff --git a/pointcloud_octomap_updater/src/pointcloud_octomap_updater.cpp b/pointcloud_octomap_updater/src/pointcloud_octomap_updater.cpp
index 7ab437a75a..0f28d369a0 100644
--- a/pointcloud_octomap_updater/src/pointcloud_octomap_updater.cpp
+++ b/pointcloud_octomap_updater/src/pointcloud_octomap_updater.cpp
@@ -120,7 +120,7 @@ bool PointCloudOctomapUpdater::initialize(const rclcpp::Node::SharedPtr& node)
   node_ = node;
   tf_buffer_ = std::make_shared<tf2_ros::Buffer>(node_->get_clock());
   auto create_timer_interface =
-      std::make_shared<tf2_ros::CreateTimerROS>(node->get_node_base_interface(), node->get_node_timers_interface());
+      std::make_shared<tf2_ros::CreateTimerROS>(tf2_ros::CreateTimerROS::RequiredInterfaces(*node));
   tf_buffer_->setCreateTimerInterface(create_timer_interface);
   tf_listener_ = std::make_shared<tf2_ros::TransformListener>(*tf_buffer_);
   shape_mask_ = std::make_unique<point_containment_filter::ShapeMask>();
