diff --git a/include/warehouse_ros/transform_collection.h b/include/warehouse_ros/transform_collection.h
index ee5a80d..a50cbb0 100644
--- a/include/warehouse_ros/transform_collection.h
+++ b/include/warehouse_ros/transform_collection.h
@@ -113,7 +113,7 @@ public:
     rclcpp::Clock::SharedPtr clock = std::make_shared<rclcpp::Clock>(RCL_SYSTEM_TIME);
     tf_buffer_ = std::make_shared<tf2_ros::Buffer>(clock);
-    tf_ = std::make_shared<tf2_ros::TransformListener>(*tf_buffer_, node_, false);
+    tf_ = std::make_shared<tf2_ros::TransformListener>(*tf_buffer_, *node_, false);
   }
 
   /// Will return the transform if it becomes available before the timeout
