diff --git a/rosidl_typesupport_microxrcedds_c/cmake/rosidl_typesupport_microxrcedds_c_generate_interfaces.cmake b/rosidl_typesupport_microxrcedds_c/cmake/rosidl_typesupport_microxrcedds_c_generate_interfaces.cmake
index b97d4cd..1211f4b 100644
--- a/rosidl_typesupport_microxrcedds_c/cmake/rosidl_typesupport_microxrcedds_c_generate_interfaces.cmake
+++ b/rosidl_typesupport_microxrcedds_c/cmake/rosidl_typesupport_microxrcedds_c_generate_interfaces.cmake
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-find_package(ament_cmake_ros REQUIRED)
+find_package(ament_cmake_ros_core REQUIRED)
 find_package(microcdr REQUIRED)
 find_package(rmw REQUIRED)
 
@@ -196,6 +196,8 @@ if(NOT rosidl_generate_interfaces_SKIP_INSTALL)
   install(
     TARGETS
       ${rosidl_generate_interfaces_TARGET}${_target_suffix}
+    EXPORT
+      ${rosidl_generate_interfaces_TARGET}${_target_suffix}
     ARCHIVE DESTINATION
       lib
     LIBRARY DESTINATION
@@ -204,9 +206,10 @@ if(NOT rosidl_generate_interfaces_SKIP_INSTALL)
       bin
     )
 
-  rosidl_export_typesupport_libraries(${_target_suffix}
+  rosidl_export_typesupport_targets(${_target_suffix}
     ${rosidl_generate_interfaces_TARGET}${_target_suffix}
     )
+  ament_export_targets(${rosidl_generate_interfaces_TARGET}${_target_suffix})
 endif()
 
 
diff --git a/rosidl_typesupport_microxrcedds_c/package.xml b/rosidl_typesupport_microxrcedds_c/package.xml
index 07a6832..f5dbea9 100644
--- a/rosidl_typesupport_microxrcedds_c/package.xml
+++ b/rosidl_typesupport_microxrcedds_c/package.xml
@@ -8,12 +8,12 @@
   <license>Apache License 2.0</license>
   <author>Borja Outerelo</author>
 
-  <buildtool_depend>ament_cmake_ros</buildtool_depend>
+  <buildtool_depend>ament_cmake_ros_core</buildtool_depend>
   <buildtool_depend>microcdr</buildtool_depend>
   <buildtool_depend>rosidl_cmake</buildtool_depend>
   <buildtool_depend>rosidl_runtime_c</buildtool_depend>
 
-  <buildtool_export_depend>ament_cmake_ros</buildtool_export_depend>
+  <buildtool_export_depend>ament_cmake_ros_core</buildtool_export_depend>
   <buildtool_export_depend>microcdr</buildtool_export_depend>
   <buildtool_export_depend>python3</buildtool_export_depend>
   <buildtool_export_depend>rosidl_cmake</buildtool_export_depend>
