diff --git a/CMakeLists.txt b/CMakeLists.txt
index b0a432a..34e040f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,7 +29,7 @@ find_package(image_transport REQUIRED)
 find_package(sensor_msgs REQUIRED)
 find_package(geometry_msgs REQUIRED)
 find_package(cv_bridge REQUIRED)
-find_package(Qt5Widgets REQUIRED)
+find_package(Qt6 REQUIRED COMPONENTS Widgets)
 find_package(ament_cmake_python REQUIRED)
 
 set(rqt_image_view_SRCS
@@ -46,9 +46,9 @@ set(rqt_image_view_UIS
   src/rqt_image_view/image_view.ui
 )
 
-qt5_wrap_cpp(rqt_image_view_MOCS ${rqt_image_view_HDRS})
+qt6_wrap_cpp(rqt_image_view_MOCS ${rqt_image_view_HDRS})
 
-qt5_wrap_ui(rqt_image_view_UIS_H ${rqt_image_view_UIS})
+qt6_wrap_ui(rqt_image_view_UIS_H ${rqt_image_view_UIS})
 
 add_library(${PROJECT_NAME} SHARED
   ${rqt_image_view_SRCS}
@@ -67,7 +67,7 @@ target_link_libraries(${PROJECT_NAME} PUBLIC
   image_transport::image_transport
   ${sensor_msgs_TARGETS}
   ${geometry_msgs_TARGETS}
-  Qt5::Widgets
+  Qt6::Widgets
 )
 target_link_libraries(${PROJECT_NAME} PRIVATE
   cv_bridge::cv_bridge)
diff --git a/src/rqt_image_view/image_view.cpp b/src/rqt_image_view/image_view.cpp
index 8ecafa6..c4b59b8 100644
--- a/src/rqt_image_view/image_view.cpp
+++ b/src/rqt_image_view/image_view.cpp
@@ -43,6 +43,8 @@
 #include <QFileDialog>  // NOLINT
 #include <QMessageBox>  // NOLINT
 #include <QPainter>  // NOLINT
+#include <QRegularExpression>  // NOLINT
+#include <QRegularExpressionValidator>  // NOLINT
 
 namespace rqt_image_view
 {
@@ -121,8 +123,9 @@ void ImageView::initPlugin(qt_gui_cpp::PluginContext & context)
   ui_.image_frame->setOuterLayout(ui_.image_layout);
 
   // see http://www.ros.org/wiki/ROS/Concepts#Names.Valid_Names (but also accept an empty field)
-  QRegExp rx("([a-zA-Z/][a-zA-Z0-9_/]*)?");
-  ui_.publish_click_location_topic_line_edit->setValidator(new QRegExpValidator(rx, this));
+  QRegularExpression rx("([a-zA-Z/][a-zA-Z0-9_/]*)?");
+  ui_.publish_click_location_topic_line_edit->setValidator(
+    new QRegularExpressionValidator(rx, this));
   connect(ui_.publish_click_location_check_box, SIGNAL(toggled(bool)), this,
       SLOT(onMousePublish(bool)));
   connect(ui_.image_frame, SIGNAL(mouseLeft(int,int)), this, SLOT(onMouseLeft(int,int)));  // NOLINT
