diff --git a/CMakeLists.txt b/CMakeLists.txt
index ce013e59d9..1d268e888c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,9 +43,13 @@ target_include_directories(
   moveit_setup_app_plugins
   PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
          $<INSTALL_INTERFACE:include/moveit_setup_app_plugins>)
-ament_target_dependencies(
-  moveit_setup_app_plugins ament_index_cpp moveit_ros_visualization
-  moveit_setup_framework pluginlib rclcpp)
+target_link_libraries(
+  moveit_setup_app_plugins
+  ament_index_cpp::ament_index_cpp
+  moveit_ros_visualization::moveit_ros_visualization
+  moveit_setup_framework::moveit_setup_framework
+  pluginlib::pluginlib
+  rclcpp::rclcpp)
 
 if(BUILD_TESTING)
   find_package(ament_cmake_gtest REQUIRED)
