diff --git a/CMakeLists.txt b/CMakeLists.txt
index da65c8f..380b187 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,9 +53,11 @@ target_compile_definitions(ublox_dgnss_components
   PRIVATE "UBLOX_DGNSS_NODE_BUILDING_DLL"
 )
 
+if(NOT APPLE)
 target_link_options(ublox_dgnss_components PRIVATE
   "LINKER:--allow-multiple-definition"
 )
+endif()
 
 target_link_libraries(ublox_dgnss_components PUBLIC
   ${rtcm_msgs_TARGETS}
