diff --git a/src/motion_capture_tracking_node.cpp b/src/motion_capture_tracking_node.cpp
index ace9626..5d99a63 100644
--- a/src/motion_capture_tracking_node.cpp
+++ b/src/motion_capture_tracking_node.cpp
@@ -199,7 +199,7 @@ int main(int argc, char **argv)
   tracker.setLogWarningCallback(std::bind(logWarn, node->get_logger(), std::placeholders::_1));
 
   // prepare TF broadcaster
-  tf2_ros::TransformBroadcaster tfbroadcaster(node);
+  tf2_ros::TransformBroadcaster tfbroadcaster(*node);
   std::vector<geometry_msgs::msg::TransformStamped> transforms;
 
   pcl::PointCloud<pcl::PointXYZ>::Ptr markers(new pcl::PointCloud<pcl::PointXYZ>);
