diff --git a/CMakeLists.txt b/CMakeLists.txt
index 03c58b9..56b2afc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,6 +6,7 @@ find_package(ros2_control_cmake REQUIRED)
 
 # find dependencies
 find_package(ament_cmake REQUIRED)
+find_package(ament_index_cpp REQUIRED)
 find_package(controller_manager REQUIRED)
 find_package(Eigen3 REQUIRED)
 find_package(fmt REQUIRED)
@@ -179,6 +180,7 @@ PUBLIC
   control_toolbox::control_toolbox
   Eigen3::Eigen
 PRIVATE
+  ament_index_cpp::ament_index_cpp
   mujoco_simulate
   Threads::Threads
   lodepng
diff --git a/src/mujoco_system_interface.cpp b/src/mujoco_system_interface.cpp
index 7364f44..3114709 100644
--- a/src/mujoco_system_interface.cpp
+++ b/src/mujoco_system_interface.cpp
@@ -46,7 +46,7 @@
 #include <std_msgs/msg/string.hpp>
 #include <unordered_map>
 
-#include <ament_index_cpp/get_package_share_directory.hpp>
+#include <ament_index_cpp/get_package_share_path.hpp>
 #if !ROS_DISTRO_HUMBLE
 #include <hardware_interface/helpers.hpp>
 #endif
@@ -785,7 +785,8 @@ MujocoSystemInterface::on_init(const hardware_interface::HardwareComponentInterf
 
       // Add ros2 control icon for the taskbar
       std::string icon_location =
-          ament_index_cpp::get_package_share_directory("mujoco_ros2_control") + "/resources/mujoco_logo.png";
+          (ament_index_cpp::get_package_share_path("mujoco_ros2_control") / "resources/mujoco_logo.png")
+              .string();
       std::vector<unsigned char> image;
       unsigned width, height;
       unsigned error = lodepng::decode(image, width, height, icon_location);
