diff --git a/CMakeLists.txt b/CMakeLists.txt
index 99a8ccc..b59e949 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,6 @@
 # judge which cmake codes to use 
+set(ROS_EDITION "ROS2")
+set(HUMBLE_ROS "humble")
 if(ROS_EDITION STREQUAL "ROS1")
 
   # Copyright(c) 2019 livoxtech limited.
@@ -190,6 +192,7 @@ else(ROS_EDITION STREQUAL "ROS2")
 
   cmake_minimum_required(VERSION 3.14)
   project(livox_ros_driver2)
+  configure_file(${CMAKE_CURRENT_SOURCE_DIR}/package_ROS2.xml ${CMAKE_CURRENT_SOURCE_DIR}/package.xml)
 
   # Default to C99
   if(NOT CMAKE_C_STANDARD)
diff --git a/package_ROS2.xml b/package_ROS2.xml
index 8169adc..1577297 100644
--- a/package_ROS2.xml
+++ b/package_ROS2.xml
@@ -17,7 +17,6 @@
   <depend>rclcpp_components</depend>
   <depend>std_msgs</depend>
   <depend>sensor_msgs</depend>
   <depend>rcutils</depend>
-  <depend>pcl_conversions</depend>
   <depend>rcl_interfaces</depend>
   <depend>libpcl-all-dev</depend>
diff --git a/src/include/ros2_headers.h b/src/include/ros2_headers.h
index 9104486..c8ff1cd 100644
--- a/src/include/ros2_headers.h
+++ b/src/include/ros2_headers.h
@@ -28,7 +28,8 @@
 #include <future>
 
 #include <rclcpp/rclcpp.hpp>
-#include <pcl_conversions/pcl_conversions.h>
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
 #include <sensor_msgs/msg/point_cloud2.hpp>
 #include <sensor_msgs/msg/imu.hpp>
 #include "livox_ros_driver2/msg/custom_point.hpp"
