diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3d26d03f58..1dff23e7f4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,15 +45,15 @@ add_library(
 target_include_directories(
   ${PROJECT_NAME} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
                          $<INSTALL_INTERFACE:include/moveit_setup_core_plugins>)
-ament_target_dependencies(
+target_link_libraries(
   ${PROJECT_NAME}
-  ament_index_cpp
-  moveit_ros_visualization
-  moveit_setup_framework
-  pluginlib
-  rclcpp
-  srdfdom
-  urdf)
+  ament_index_cpp::ament_index_cpp
+  moveit_ros_visualization::moveit_ros_visualization
+  moveit_setup_framework::moveit_setup_framework
+  pluginlib::pluginlib
+  rclcpp::rclcpp
+  srdfdom::srdfdom
+  urdf::urdf)
 
 install(
   TARGETS moveit_setup_core_plugins
